src/java/org/apache/cassandra/io/sstable/SSTableWatcher.java

```diff
@@ -53,4 +65,43 @@ default Set<Component> discoverComponents(Descriptor descriptor, Set<Component>
     {
         return existing;
     }
+
+    /**
+     * Appends new component names to the TOC component.
+     */
+    default void appendTOC(Descriptor descriptor, Collection<Component> components)
```

Comment:
I'm actually not quite sure `SSTableWatcher` is the right place for this: the interface is called "watcher" and is about discovering components when opening sstables, so having TOC writing methods feels a bit out of place. Maybe that just means we need to rename/update descriptions to expand what this interface is about, I'm not sure. I'll have some related comments on the CNDB side, which are maybe a bit more important, but figured I'd point this out here too.